text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+Additionally, `<lwc.dynamic>` is intended to be used in tandem with `lwc:is`, otherwise `<lwc:dynamic>` serves no purpose. A warning should be issued when `<lwc.dynamic>` is used without the `lwc:is` template directive.
+
+##### `lwc:is` directive
+
+A new template directive called `lwc:is` will be used to signal to compiler that the constructor is unknown at compile time. 
+
+The directive has the following properties:
+
+1. It can only be used with `<lwc.dynamic>` tag.
+    - An error will be issued when `lwc:is` is used on any other element.
+2. At compile time the value supplied to `lwc:is` can only be an expression.
+3. At runtime the expression supplied to `lwc:is` must be a `LightningElementConstructor`.
+    - An error will be thrown if anything other than a `LightningElementConstructor` is passed to `lwc:is`.
+
+#### Instantiation of the custom element
+
+Dynamic component instantiation follows the same creation path as any other LWC custom element. When the dynamic component is ready to be rendered, the element will be created and mounted to the DOM.  If there is an existing element supplied to `lwc:is`, it will be unmounted as well.
+
+This means that when the value provided to `lwc:is` changes the component and all of its children will be destroyed and recreated using the new constructor.  In addition, when the value provided to `lwc:is` is evaluated, the LWC engine will render depending on the following scenarios:
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc.dynamic>`. The tag name used for the dynamic component is the value of `lightningElementName` on `LightningElementConstructor`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+It is important to note that `<lwc.dynamic>` will not be rendered to the DOM.  As such, it is not available to be selected using `this.template.querySelector`.  The actual custom element must be selected once it has been rendered to the DOM.  The dynamic component can be selected either through the `lightningElementName` property on the constructor or through another attribute assigned to the component.
+
+Component authors can use the `connectedCallback` of the dynamic component to signal when the component has been rendered to the DOM.
```

Comment:
Here are a couple of other options to detect whenever a dynamic component is mounted:
- Once a dynamic component constructor is set, it is guaranteed that will be mounted in the next rendering cycle. The parent component can wait until `renderedCallback` in the parent component after the component constructor is set to detect when it gets mounted.
- Use a `MutationObserver`.

Reply:
Thanks for these, added them to the [RFC](https://github.com/salesforce/lwc-rfcs/pull/71/files#diff-a1de05148a61ea9b00f70bd4ff5feb3895c16dd99707b173ba57c081b859250bR193-R194)!